src/server/api/errors.js

```diff
@@ -22,27 +19,24 @@ export async function accessRequired(user, orgId, role, allowSuperadmin = false)
     return
   }
   // require a permission at-or-higher than the permission requested
-  const acceptableRoles = accessHierarchy.slice(accessHierarchy.indexOf(role))
-  const hasRole = await userHasRole(user.id, orgId, acceptableRoles)
+  const hasRole = await cacheableData.user.userHasRole(user, orgId, role)
   if (!hasRole) {
     throw new GraphQLError('You are not authorized to access that resource.')
   }
 }
 
-export async function assignmentRequired(user, assignmentId) {
+export async function assignmentRequired(user, assignmentId, assignment) {
   authRequired(user)
 
   if (user.is_superadmin) {
     return
   }
-
-  const [assignment] = await r.knex('assignment')
-  .where({
-    user_id: user.id,
-    id: assignmentId
-  }).limit(1)
-
-  if (typeof assignment === 'undefined') {
+  if (assignment && assignment.user_id === user.id) {
+    // if we are passed the full assignment object, we can test directly
+    return
+  }
+  const userHasAssignment = await cacheableData.assignment.hasAssignment(user.id, assignmentId)
```

Comment:
cacheableData.assignment is comming up as undefined (so the texter texting view is coming up as blank ex: `http://localhost:3000/app/1/todos/28/text`)

Reply:
fixed with a test